app/components/apply-page/app-timeline.tsx

```diff
@@ -0,0 +1,92 @@
+import { useState, useEffect, useCallback } from "react";
+import { Progress } from "../ui/progress";
+
+export interface Period {
+  start: Date;
+  end: Date;
+}
+
+// to mark the 3 periods, one for applications opening, one for interview, one for application decisions
+interface AppTimelineProps {
+  periods: [Period, Period, Period];
+}
+
+const AppTimeline = ({ periods }: AppTimelineProps) => {
+  const [progress, setProgress] = useState<number>(0);
+
+  const startDate = periods[0].start;
+  const endDate = periods[2].end;
+
+  const calculateProgress = useCallback(
+    (current: Date): number => {
+      const total = endDate.getTime() - startDate.getTime();
+      const elapsed = current.getTime() - startDate.getTime();
+      return Math.min(Math.max((elapsed / total) * 100, 0), 100);
+    },
+    [startDate, endDate],
+  );
+
+  const formatDate = (date: Date): string => {
+    return date.toLocaleDateString("en-US", {
+      month: "numeric",
+      day: "numeric",
+      year: "2-digit",
+    });
+  };
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      const now = new Date();
+      setProgress(calculateProgress(now));
+    }, 1000);
+
+    return () => clearInterval(timer);
+  }, [calculateProgress]);
+
+  const labels = ["Applications Open", "Interviews", "Application Decisions"];
+
+  return (
+    <div className="w-full">
+      <div className="relative">
+        <div className="relative h-4">
+          <Progress value={progress} className="absolute w-full h-4" />
+
+          {periods.map((period, index) => {
+            const position = `${(index / 2) * 100}%`;
+            return (
+              <div
+                key={index}
+                className="absolute top-0 transform -translate-x-1/2"
```

Comment:
whats the purpose of the top-0 and the transform? Should it be a y transform to center the dots instead of an x transform?